src/client/Microsoft.Identity.Client/Cache/CacheMissReason.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Enum describing the reason for cache refresh and fetching access token from ESTS.
+    /// </summary>
+    public enum CacheMissReason
+    {
+        /// <summary>
+        /// When the cache is not supposed to be hit to make the request (interactive call, username password call, device code flow, etc.)
+        /// </summary>
+        None = 0,
+        /// <summary>
+        /// When the token request goes to the identity provider because force_refresh was set to true
+        /// </summary>
+        ForceRefresh = 1,
+        /// <summary>
+        /// When the token request goes to the identity provider because no cached access token exists
+        /// </summary>
+        NoCachedAT = 2,
```

Comment:
Since it's public now, maybe spell it out - `NoCachedToken`, `NoCachedAccessToken`?